### ISSUE TITLE
Question about converting save files containing RTC data

### ISSUE THREAD

**Issue by euan-forrester** [MiSTer-devel/GBA_MiSTer#138]
Hello, I'm the author of savefileconverter.com

I recently had a user mention that they were unable to convert a Pokemon Emerald save file to work with the MiSTer GBA core. It seems that this specific game wants a save file that's `0x22000` bytes long, instead of the usual `0x20000` bytes. 

- Is the extra data RTC information? 
- Looking at the file (and from the information that attempting to load a `0x20000` byte long save file doesn't work), it looks like the extra `0x2000` bytes are prepended to the beginning of the file, rather than appended to the end?
- For the savefileconverter site, can I just prepend `0x2000` bytes of `0x00` to the beginning of a raw save file to get it to work?
- Are there games with RTC data (or any other kind of extra data) in their saves that's a different length than `0x2000` bytes?
- Is there a way that I can correctly initialize RTC data using the brower's current time? Or should I only fill it with `0x00`?

Thanks very much for any information you're able to give me!

**Comment by RobertPeip** [comment]
I cannot see why the RTC data would be at the beginning, because it's at the end. (file is attached)
[Pokemon - Smaragd-Edition (Germany).zip](https://github.com/MiSTer-devel/GBA_MiSTer/files/13461091/Pokemon.-.Smaragd-Edition.Germany.zip)

You can just generate the file with the normal size without extra data and it can be loaded, it behaves the same as adding additional zeros at the end.

Unfortunatly you cannot just add browser time, because the RTC format is the internal time of the game + the offset at point of turning mister/game off so when turned back on it can count up to the new time.
So best is really just use raw data and leave RTC away.

**Comment by euan-forrester** [comment]
Yeah, that was the first thing I told him: to just load a regular raw save file and he said it didn't work but he'd gotten other games to work so I assumed he didn't make any mistakes with the file.

Then he gave me a sample file, which was `0x22000` bytes long. But I see yours is `0x20200` bytes. And yeah, the RTC data is much more obvious in yours. 

It also makes much more sense for the core to append the RTC data to the end than to the beginning of the file.

Okay so it seems that this user is just confused. I thought that I had vetted him enough before coming here, but clearly I didn't so I apologize. Thanks for the help, and the confirmation that everything works the way I thought it would!

**Comment by RobertPeip** [comment]
No worries, if you have any question, don't hesitate to ask and thank you for your work!

**Comment by euan-forrester** [comment]
Thanks very much! You've been really helpful and kind in our interactions and in ones I've seen with other people and I really appreciate it!
